src/qt/walletcontroller.cpp

```diff
@@ -99,6 +99,9 @@ WalletModel* WalletController::getOrCreateWallet(std::unique_ptr<interfaces::Wal
 
     // Instantiate model and register it.
     WalletModel* wallet_model = new WalletModel(std::move(wallet), m_node, m_platform_style, m_options_model, nullptr);
+    // Handler callback runs in a different thread so fix wallet model thread affinity.
+    wallet_model->moveToThread(thread());
+    wallet_model->setParent(this);
```

Comment:
Exactly, can't set parent on constructor because parent thread can be different.

`setParent` is not necessary at the moment because (afaict) the wallet is always unloaded here:
https://github.com/bitcoin/bitcoin/blob/8046a3e0befeea641b6309bc0c742b7481e681d9/src/qt/walletcontroller.cpp#L104

However I think it's good habit to set ownership, so at least the following would release all models:
https://github.com/bitcoin/bitcoin/blob/8046a3e0befeea641b6309bc0c742b7481e681d9/src/qt/bitcoin.cpp#L216

Reply:
PR description has been updated.